Automattic/wp-calypso
# Jetpack Settings: Add jetpack module activation reducers, actions types, action creators and selectors

Part of #9171 

This PR is a kick-off for implementing Jetpack Settings and Module related operation in the redux state tree.

#### Changes proposed in this Pull Request:

* Implements `client/state/jetpack-settings/reducer.js`
  * Introduces a `jetpackSettings` leaf on the redux state tree.
  * Enables this reducer in `client/state/index.js`.
* Implements `client/state/jetpack-settings/modules/reducer.js`. _The `jetpack-settings/reducer` combines this reducer_. 
  * Introduces the `jetpackSettings.jetpackModules` leaf in the state tree.
* Introduces an action creator `activateModule` for activating a jetpack module.
  * It's based on a non-existant `lib/wp.undocumented().activateJetpackModule()` method. _The action tests mock this method for passing_.
* Introduces the selectors `isActivatingModule()` and `isModuleActivated` for querying the state.

#### Testing instructions

1. Run 
    ```
    $ npm run test-client client/state/jetpack-settings/
    ```
1. Expect the tests to pass.
